warehouse/oidc/views.py

```diff
@@ -0,0 +1,86 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+
+from pyramid.view import view_config
+from sqlalchemy import func
+
+from warehouse.macaroons.interfaces import IMacaroonService
+from warehouse.oidc.interfaces import IOIDCProviderService
+from warehouse.packaging.models import Project
+
+
+@view_config(
+    route_name="oidc.mint_token",
+    require_methods=["POST"],
+    renderer="json",
+    require_csrf=False,
+    has_translations=False,
+)
+def mint_token_from_oidc(request):
+    def _invalid(msg):
+        request.response.status = 422
+        return {"error": msg}
```

Comment:
It might be nice for these to both have something short and unique to key an error off of, as well as a longer human-readable explanation of why this failed.

Reply:
Make sense. Something like this?

```
{
  success: bool,
  description: Optional[str],
}
```

Reply:
Maybe something like:

```
{
   "message": "Token request failed",
   "errors": [
     {
       "code": "some_unique_string",
       "description: "A nice human-readable description"
     }
   ]
 }
```